src/MeetingsSDKAdapter.js

```diff
@@ -679,36 +678,36 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
     }
 
     const enableSharingStream = async () => {
-      const [, localShare] = await sdkMeeting.getMediaStreams({sendShare: true});
-
-      this.meetings[ID].localShare = localShare;
-
       // Due to SDK limitation around local media updates,
       // we need to emit a custom event for video mute updates
       sdkMeeting.emit(EVENT_MEDIA_LOCAL_UPDATE, {
         control: SHARE_CONTROL,
         state: MeetingControlState.DISABLED,
       });
 
+      const [, localShare] = await sdkMeeting.getMediaStreams({sendShare: true});
+
+      this.meetings[ID].localShare = localShare;
+
       await sdkMeeting.updateShare({stream: localShare, sendShare: true, receiveShare: true});
     };
 
     const disableSharingStream = async () => {
       // TODO (SPARK-148910): emit should be at the end of the function.
```

Comment:
"emit should be at the end..." - not anymore, you can remove this.

you can leave the comment about the call taking 30s, maybe something like:

// disable the control while the SDK is stopping sharing, which make take 30 seconds or more